detectors/resources/src/main/java/com/google/cloud/opentelemetry/detectors/GAEResource.java

```diff
@@ -0,0 +1,92 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+public class GAEResource implements ResourceProvider {
+  private static final String GAE_ENVIRONMENT = "GAE_ENV";
+  private final GCPMetadataConfig metadata;
+  private final EnvVars envVars;
+
+  public GAEResource() {
+    this.metadata = GCPMetadataConfig.DEFAULT_INSTANCE;
+    this.envVars = EnvVars.DEFAULT_INSTANCE;
+  }
+
+  // For testing only
+  GAEResource(GCPMetadataConfig metadata, EnvVars envVars) {
+    this.metadata = metadata;
+    this.envVars = envVars;
+  }
+
+  public Attributes getAttributes() {
+    if (!metadata.isRunningOnGcp()) {
+      return Attributes.empty();
+    }
+
+    AttributesBuilder attrBuilders = Attributes.builder();
+    attrBuilders.put(ResourceAttributes.CLOUD_PROVIDER, ResourceAttributes.CloudProviderValues.GCP);
+
+    if (envVars.get("GAE_SERVICE") != null) {
+      // add the resource attributes for App Engine
+      attrBuilders.put(
+          ResourceAttributes.CLOUD_PLATFORM, ResourceAttributes.CloudPlatformValues.GCP_APP_ENGINE);
+
+      String appModuleName = envVars.get("GAE_SERVICE");
+      if (appModuleName != null) {
+        attrBuilders.put(ResourceAttributes.FAAS_NAME, appModuleName);
+      }
+
+      String appVersionId = envVars.get("GAE_VERSION");
+      if (appVersionId != null) {
+        attrBuilders.put(ResourceAttributes.FAAS_VERSION, appVersionId);
+      }
+
+      String appInstanceId = envVars.get("GAE_INSTANCE");
+      if (appInstanceId != null) {
+        attrBuilders.put(ResourceAttributes.FAAS_ID, appInstanceId);
+      }
+
+      updateAttributesWithRegion(attrBuilders);
+    }
+    return attrBuilders.build();
+  }
+
+  @Override
+  public Resource createResource(ConfigProperties config) {
+    return Resource.create(getAttributes());
+  }
+
+  /**
+   * Selects the correct method to extract the region, depending on the GAE environment.
+   *
+   * @param attributesBuilder The {@link AttributesBuilder} object to which the extracted region
+   *     would be added.
+   */
+  private void updateAttributesWithRegion(AttributesBuilder attributesBuilder) {
+    if (envVars.get(GAE_ENVIRONMENT) != null && envVars.get(GAE_ENVIRONMENT).equals("standard")) {
+      AttributesExtractorUtil.addCloudRegionFromMetadataUsingRegion(attributesBuilder, metadata);
+    } else {
+      AttributesExtractorUtil.addCloudRegionFromMetadataUsingZone(attributesBuilder, metadata);
```

Comment:
Should this add availability_zone as well, since we have that information?

Reply:
We can add them, but as per the spec document, `availability_zone` is not required. 

No prior resources detections that were added (Cloud Functions, Cloud Run & App Engine) included any additional attributes other than the ones mentioned in the spec.

Reply:
this is fine for now

Reply:
Cool, I think this will be an easy change to make in the future, but it would be better if we first prioritize resolving https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/issues/174 before making changes too the  attributes.